packages/microsite/src/scripts/dev.tsx

```diff
@@ -0,0 +1,88 @@
+import { startDevServer, loadAndValidateConfig } from 'snowpack';
+import { resolve } from 'path';
+import type { IncomingMessage, ServerResponse } from 'http';
+import polka from 'polka';
+
+const pageScript = (page: string) => `import { csr } from '/web_modules/microsite/runtime/csr.js';
+import Page from '${page}';
+csr(Page);`;
+
+const globalScript = () => `(async () => {
+    try {
+        const global = await import('/src/global.js').then(mod => mod.default);
+        if (global) global();
+    } catch (e) {}
+    try { await import('/src/global.css.proxy.js'); } catch (e) {}
+
+    document.documentElement.style.removeProperty('visibility');
+})()`;
+
+const doc = (page: string) => `<!doctype html>\n<!-- Generated by microsite -->\n<html lang="en" dir="ltr" style="visibility:hidden">\n\t<head></head>\n\t<body>
+\t\t<div id="__microsite"></div>
+\t\t<script data-csr="true">window.HMR_WEBSOCKET_URL = 'ws://localhost:3333';</script>
+\t\t<script type="module" src="/__snowpack__/hmr-client.js"></script>
+\t\t<script type="module" data-microsite="page">${pageScript(page)}</script>
+\t\t<script type="module" data-microsite="global">${globalScript()}</script>
+</body>\n</html>`
+
+export async function dev() {
+    const cwd = process.cwd();
+    const config = loadAndValidateConfig({ config: resolve(import.meta.url.slice('file:'.length), '../../snowpack.config.cjs') }, {});
```

Comment:
`loadAndValidateConfig` will be deprecated in the next Snowpack v3.0 RC in favor of the much more straightforward `createConfiguration`. You could implement this today like this:

```js
import {createConfiguration} from 'snowpack';
const userConfigLocation = resolve(import.meta.url.slice('file:'.length), '../../snowpack.config.cjs');
const userConfig = JSON.parse(fs.readFileSync(userConfigLocation, 'utf8'));
const config = createConfiguration(userConfig);
```

A helper `loadConfiguration(root: string)` function could also make sense, which would make it easier to support snowpack config files in any format. Don't have the time to add this right now, but lmk if that would be interesting and I can get it in in time for v3

Reply:
Thanks so much for taking the time to review! 

I did see that `createConfiguration` was included in the types for `snowpack@3.0.0-rc.1`, but it wasn't actually exported in the ESM entrypoint. Looks like [that has been fixed](https://github.com/snowpackjs/snowpack/blob/3eddb91d8cfe68caf6da047be0efb99cbb8c5813/snowpack/index.esm.mjs#L6)! 🎉 

`loadConfiguration(root: string)` would be nice to have, but not a blocker. It would enable support for _optional_ custom `snowpack.config.js` files (custom plugins, etc) in Microsite projects.